### DISCUSSION TITLE
Does watch feature implementation strongly depend on memory-storage?

### DISCUSSION THREAD

**Discussion by qmloong** [clusterpedia-io/clusterpedia#433] [Q&A] [answered]
I want to know if I need to use the `watch` feature, then do I must use the memory as a storage layer?

related issue #265

**Comment by Iceber** [comment, marked as answer]
For the present, yes indeed.

But we are also looking for a more general Watch implementation.

Clusterpedia actually accesses the storage component through the storage layer interface and does not care about the specific storage layer implementation, and Clusterpedia is providing a custom storage layer based on the go plugin, so that users can provide specific Watch implementations at will.

For a generic Watch implementation, my initial thought is to base it on some lightweight message queue, such as nsq, which decouples the synchronization and query components of resources.
We can discuss and push this together.

**Reply by qmloong** [reply]
I've recently learned about clusterpedia and if I have time, I will be pleasured to take part in.
